src/main/java/hudson/plugins/filesystem_scm/RemoteFolderDiff.java

```
@@ -1,103 +1,116 @@
-
-		@Override
-		public List<FolderDiff.Entry> invoke(File workspace, VirtualChannel channel) throws IOException {
-			setDstPath(workspace.getAbsolutePath());
-			List<FolderDiff.Entry> newFiles = getNewOrModifiedFiles(lastBuildTime, false, false);
-			List<FolderDiff.Entry> delFiles = getDeletedFiles(lastSuccessfulBuildTime, false, false);
-			List<FolderDiff.Entry> files = new ArrayList<FolderDiff.Entry>();
-			files.addAll(newFiles);
-			files.addAll(delFiles);
-			return files;
-		}	
-	}
+    protected StringBuffer buf;
+    protected long lastBuildTime;
+    protected long lastSuccessfulBuildTime;
+
+    public RemoteFolderDiff() {
+        buf = new StringBuffer();
+    }
+
+    public long getLastBuildTime() {
+        return lastBuildTime;
+    }
+
+    public void setLastBuildTime(long lastBuildTime) {
+        this.lastBuildTime = lastBuildTime;
+    }
+
+    public long getLastSuccessfulBuildTime() {
+        return lastSuccessfulBuildTime;
+    }
+
+    public void setLastSuccessfulBuildTime(long lastSuccessfulBuildTime) {
+        this.lastSuccessfulBuildTime = lastSuccessfulBuildTime;
+    }
+
+    @Override
+    protected void log(String msg) {
+        buf.append(msg).append("\n");
+    }
+
+    @Override
+    protected void copyFile(File src, File dst) throws IOException {
+        FilePath srcpath = new FilePath(src);
+        FilePath dstpath = new FilePath(dst);
+        try {
+            changeDestinationPropertiesOnUnixSytem(dstpath, isUnix());
+            srcpath.copyToWithPermission(dstpath);
+        } catch (InterruptedException e) {
+            IOException ioe = new IOException();
+            ioe.initCause(e);
+            throw ioe;
+        }
+    }
+
+    private static void changeDestinationPropertiesOnUnixSytem(FilePath dstpath, boolean isUnix)
+            throws IOException, InterruptedException, PosixException {
+        // if not write-able, then we can't copy, have to set it to write-able

```

I think it is a bad idea in general (though it existed in the code for years). Maybe we will need a flag to opt-out from this behavior later

---

yeah, it might be better to fail early instead of doing platform dependent stuff. But then we should check everything at setup time.

---

The intent of this code is probably to save the user from messing around with the filesystem.